spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/BinderConfiguration.java

```diff
@@ -34,15 +34,21 @@
 
 	private final boolean inheritEnvironment;
 
+	private final boolean applicationProvided;
+
 	/**
 	 * @param binderType the binder type used by this configuration
 	 * @param properties the properties for setting up the binder
-	 * @param inheritEnvironment whether the binder should inherit the environment of the module
+	 * @param inheritEnvironment whether the binder should inherit the environment of the
+	 * module
+	 * @param applicationProvided whether the binder is user defined
```

Comment:
I think this would be more clear if it said something like:

```
whether the binder is provided by an application, as opposed to being user defined
```